src/Medidata.ZipkinTracer.Core/ZipkinConfig.cs

```diff
@@ -84,5 +84,29 @@ internal bool IsInDontSampleList(string path)
             }
             return false;
         }
+
+        /// <summary>
+        /// Try parse string to bool
+        /// </summary>
+        /// <param name="stringValue"></param>
+        /// <returns>returns true if value can be parsed to bool</returns>
+        private bool TryParseToBool(string stringValue, out bool booleanValue)
+        {
+            booleanValue = false;
+
+            if (string.IsNullOrWhiteSpace(stringValue)) return false;
+
+            switch (stringValue)
+            {
+                case "0":
+                    booleanValue = false;
+                    return true;
+                case "1":
+                    booleanValue = true;
+                    return true;
+                default:
+                    return Boolean.TryParse(stringValue, out booleanValue);
+            }
+        }
     }
```

Comment:
The requirement for this is it needs to check if stringValue is a valid bool value.
If it is a valid bool value (true/false/1/0) then  it would use that value.
If not, then it needs to do some calculation using the sample rate to get a bool value. Please checkout ShouldBeSampled() method at line 52